### ISSUE TITLE
Why was Wasm support removed?

### ISSUE THREAD

**Issue by perry-blueberry** [janhohenheim/foxtrot#299]
Looks like this had WASM support previously but it was removed. The only reason I can find is because CI took a long time. :thinking:

**Comment by janhohenheim** [comment]
In short: it was too tedious to maintain.
Long story:
I noticed that Wasm had a tendency to break regularly in surprising ways:
- Most dependencies are not explicitly tested against Wasm and tended to break randomly pn runtime when updating versions
- Anything involving shaders tended to work differently on WebGL (16 byte alignment, wgsl calls behaving slightly differently, etc.)
- The performance on Wasm fluctuated drastically between Bevy versions. Code that ran fine on one version suddenly became really laggy
- Testing native and Wasm locally took a really long time since building one invalidates the other. On my current machine (which I need to upgrade), a full build of native takes about 30 min. When I switch to Wasm, that's another 30 min. Finally I want to work again on native, which takes 30 min again.
- IDEs tend to automatically run cargo commands for native in the background, which invalidated the Wasm build
- Performance around 3D scenes, especially loading, takes too long on Wasm due to a lack of parallelism. See for example how [the Yarn Spinner demo](https://janhohenheim.itch.io/yarnspinner-rust-demo) takes to load even though it is an extremely simple game. This is bad enough to the point that I don't consider Bevy in its current state usable for 3D projects on the web (yet).

And lastly, I personally am interested in making 3D games for native platforms and not Wasm. This means that Wasm support is primarily a feature I added for others and not for myself. 
But combined with the mentioned issues, supporting Wasm meant spending a significant amount of time on something I didn't enjoy. I noticed I avoided working on Foxtrot at all, which is not a mindset I wanted to be in, so I removed Wasm support.

That said, if anyone is opened a PR that added Wasm support again, I'd merge it but wouldn't maintain or test it. Maybe add a note in the Readme that says Wasm support is experimental.
